udfs/tests/dataform_testing_framework/deploy_and_run_tests.sh

```diff
@@ -0,0 +1,277 @@
+
+  if [[ -f "${udfs_target_dir}"/definitions/"${dataset_id}"/test_cases.js ]]; then
+    # Temporarily rename test_cases.js to avoid deploying this file
+    mv "${udfs_target_dir}"/definitions/"${dataset_id}"/test_cases.js \
+      "${udfs_target_dir}"/definitions/"${dataset_id}"/test_cases.js.ignore
+  else
+    printf "No test_cases.js file found.\n"
+  fi
+
+  replace_js_udf_bucket_placeholder "${udfs_target_dir}"/definitions/"${dataset_id}" "${js_bucket}"
+  generate_dataform_config_and_creds "${project_id}" "${short_dataset_id}" "${udfs_target_dir}"
+  add_symbolic_dataform_dependencies "${udfs_target_dir}"
+
+  printf "Deploying UDFs from %s using dataform run command.\n" "${udfs_source_dir}"
+  if ! dataform run "${udfs_target_dir}"; then
+    # If any error occurs, delete BigQuery testing dataset before exiting with status code 1
+    bq --project_id "${project_id}" rm -r -f --dataset "${dataset_id}"
+    printf "FAILURE: Encountered an error when deploying UDFs in dataset: %s\n\n" "${dataset_id}"
+    exit 1
+  fi
+
+  if [[ -f "${udfs_target_dir}"/definitions/"${dataset_id}"/test_cases.js.ignore ]]; then
+    # Restore test_cases.js once UDFs are deployed
+    mv "${udfs_target_dir}"/definitions/"${dataset_id}"/test_cases.js.ignore \
+      "${udfs_target_dir}"/definitions/"${dataset_id}"/test_cases.js
+  fi
+}
+
+#######################################
+# description
+# Arguments:
+#   1
+#   2
+#   3
+#   4
+#######################################
+test_udfs() {
+  local project_id=$1
+  local dataset_id=$2
+  local udfs_source_dir=$3
+  local udfs_target_dir=$4
+  mkdir -p "${udfs_target_dir}"/definitions
+  # Only run Dataform tests if a test_cases.js exists
+  if [[ -f "${udfs_source_dir}"/test_cases.js ]]; then
+    cp "${udfs_source_dir}"/test_cases.js "${udfs_target_dir}"/definitions/test_cases.js
+    generate_dataform_config_and_creds "${project_id}" "${dataset_id}" "${udfs_target_dir}"
+    add_symbolic_dataform_dependencies "${udfs_target_dir}"
+    if ! dataform test "${udfs_target_dir}"; then
+      # If any error occurs, delete BigQuery testing dataset before exiting with status code 1
+      bq --project_id "${project_id}" rm -r -f --dataset "${dataset_id}"
+      rm -rf "${dataset_id}"_test
+      printf "FAILURE: Encountered an error when running UDF tests for dataset: %s\n\n" "${dataset_id}"
+      exit 1
+    fi
+  else
+    printf "Skipping Dataform unit tests since no test_cases.js file found.\n"
+  fi
+}
+
+#######################################
+# This is the main entry point of this bash script.
+# This function orchestrates the entire UDF unit testing
+# process.
+# Globals:
+#   BQ_LOCATION
+#   JS_BUCKET
+#   PROJECT_ID
+#   SHORT_SHA
+# Arguments:
+#  None
+#######################################
+main() {
+  # For now, this build script assumes all BigQuery environments
+  # live in the same location which you specify below.
+  export BQ_LOCATION=US
```

Comment:
Can we make this default to US but allow for overrides? Currently someone would have to modify the script to use a different location.